azure_functions_worker/bindings/meta.py

```diff
@@ -94,5 +104,39 @@ def to_outgoing_proto(binding: str, obj: typing.Any, *,
             f'unable to encode outgoing TypedData: '
             f'unsupported type "{binding}" for '
             f'Python type "{type(obj).__name__}"')
+    return datum
+
 
+def to_outgoing_proto(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type]) -> protos.TypedData:
+    datum = get_datum(binding, obj, pytype)
     return datumdef.datum_as_proto(datum)
+
+
+def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type],
+                      out_name: str,
+                      shmem_mgr: SharedMemoryManager) -> protos.ParameterBinding:
+    datum = get_datum(binding, obj, pytype)
+    # TODO gochaudh: IMPORTANT: Right now we set the AppSetting to disable this
+    # However that takes impact only for data coming from host -> worker
+    # Is there a way to check the AppSetting here so that this does not respond back
+    # with shared memory?
+    shared_mem_value = None
+    parameter_binding = None
+    # If shared memory is enabled, try to transfer to host over shared memory
+    if shmem_mgr.is_enabled() and shmem_mgr.is_supported(datum):
+        shared_mem_value = datumdef.Datum.to_rpc_shared_memory(datum, shmem_mgr)
+    if shared_mem_value is not None:
+        # Check if data was transferred over shared memory.
+        # If it was, then use the rpc_shared_memory field in the response message. 
+        parameter_binding = protos.ParameterBinding(
```

Comment:
I'm always torn by singleflow vs multiple returns but given that this code already does multiple returns in other places, I think we should return directly and do return None in the default case.

Reply:
But this isn't going to return None in any case - it's either `ParameterBinding` with `rpc_shared_memory` or `data`.

Reply:
True.